test/testdrive/kafka-duplicate-topic.td

```diff
@@ -0,0 +1,51 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ set schema={"type" : "record", "name" : "test", "fields": [ { "name": "f2", "type": "long" } ] }
+
+$ kafka-create-topic topic=topic0
+
+$ kafka-create-topic topic=topic1
+
+$ kafka-ingest format=avro topic=topic0 schema=${schema} publish=true repeat=1
```

Comment:
nit: maybe use different keys here, so `key1` and `key2`, say. You can use `sort-messages=true` on the later verify stanza. Or you could interleave producing to Kafka and verifying the output on the sink.

Reply:
sensible! i've made the records distinct, so we're now checking that we get one from each source.